File: RedSheet.DbEntities/Models/KnowledgeGatheringPlan.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
    [Table("KnowledgeGatheringPlans", Schema = "dbo")]
    [Serializable]
    public partial class KnowledgeGatheringPlan
    {

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public int KnowledgeGatheringPlanId { get; set; }

        [Required]
        [MaxLength(50)]
        public string KnowledgeGivenBy { get; set; }

        [Required]
        public System.DateTime KnowledgeGivenOn { get; set; }

        [Required]
        [MaxLength(250)]
        public string KnowledgeRequired { get; set; }

        [Required]
        [MaxLength(150)]
        public string Source { get; set; }

        [RelationshipTableAttribue("ProjectPowers", "dbo")]
        //Relationships
        public int ProjectPowerId { get; set; }

        [ForeignKey("ProjectPowerId")]
        public ProjectPower ProjectPower { get; set; }

        public KnowledgeGatheringPlan()
        {
        }

    }
}

File: redsheet-portal/src/app/database-models/knowledge-gathering-plan.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectPower, vKnowledgeGatheringPlanRecord } from './'
export class KnowledgeGatheringPlan {
    constructor(knowledgeGatheringPlan?: KnowledgeGatheringPlan | vKnowledgeGatheringPlanRecord) {
        let properties = ["knowledgeGatheringPlanId", "knowledgeGivenBy", "knowledgeGivenOn", "knowledgeRequired", "source", "projectPowerId",];
        for (let property of properties)
            if (knowledgeGatheringPlan && knowledgeGatheringPlan[property])
                this[property] = knowledgeGatheringPlan[property];
    }

    knowledgeGatheringPlanId: number = 0;

    @required()
    @maxLength(50)
    knowledgeGivenBy: string = undefined;

    @required()
    knowledgeGivenOn: Date = undefined;

    @required()
    @maxLength(250)
    knowledgeRequired: string = undefined;

    @required()
    @maxLength(150)
    source: string = undefined;

    @range(0, 2147483647)
    projectPowerId: number = undefined;
    projectPower: ProjectPower;


}
